website_cookieconsent/static/src/js/website_cookieconsent.js

```diff
@@ -0,0 +1,68 @@
+$(function() {
+    "use strict";
+
+    function extractLanguageFromCookie(cookieString) {
+        const cookies = cookieString.split(';');
+        let lang = null;
+
+        for (let i = 0; i < cookies.length; i++) {
+            const cookie = cookies[i].trim();
+            const [cookieName, cookieValue] = cookie.split('=');
+
+            if (cookieName === 'website_lang') {
+                let langValue = cookieValue.split('_')[0];
+                lang = langValue.toLowerCase();
+                break;
+            }
+        }
+
+        return lang;
+    }
+
+    function isValidJSON(str) {
+        try {
+            JSON.parse(str);
+            return true;
+        } catch (e) {
+            return false;
+        }
+    }
+
+    $.ajax({
+        type: 'POST',
+        url: '/website_cookieconsent',
+        dataType: 'json',
+        beforeSend: function(xhr) {
+            xhr.setRequestHeader('Content-Type', 'application/json');
```

Comment:
I think this is not needed if you use `this.rpc` as for example in [here](https://github.com/OCA/web/blob/8.0/help_online/static/src/js/help_online.js#L79).

Reply:
I think rpc is when you are on the web side. For website I used [openerp.jsonrpc](https://github.com/OCA/OCB/blob/279529c088d28b8ad58da3314a829b0b9421dff0/addons/website/static/src/js/website.js#L309)